cloudflare/resource_cloudflare_waiting_room.go

```diff
@@ -0,0 +1,223 @@
+			log.Printf("[WARN] Removing waiting room from state because it's not found in API")
+			d.SetId("")
+			return nil
+		}
+		name := d.Get("name").(string)
+		return fmt.Errorf("Error getting waiting room %q: %s", name, err)
+	}
+	d.SetId(waitingRoom.ID)
+	d.Set("name", waitingRoom.Name)
+	d.Set("description", waitingRoom.Description)
+	d.Set("suspended", waitingRoom.Suspended)
+	d.Set("host", waitingRoom.Host)
+	d.Set("path", waitingRoom.Path)
+	d.Set("queue_all", waitingRoom.QueueAll)
+	d.Set("new_users_per_minute", waitingRoom.NewUsersPerMinute)
+	d.Set("total_active_users", waitingRoom.TotalActiveUsers)
+	d.Set("session_duration", waitingRoom.SessionDuration)
+	d.Set("disable_session_renewal", waitingRoom.DisableSessionRenewal)
+	d.Set("custom_page_html", waitingRoom.CustomPageHtml)
+	return nil
+}
+
+func resourceCloudflareWaitingRoomCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+
+	newWaitingRoom := buildWaitingRoom(d)
+
+	waitingRoom, err := client.CreateWaitingRoom(context.Background(), zoneID, newWaitingRoom)
+
+	if err != nil {
+		name := d.Get("name").(string)
+		return fmt.Errorf("Error creating waiting room %q: %s", name, err)
+	}
+
+	d.SetId(waitingRoom.ID)
+
+	return resourceCloudflareWaitingRoomRead(d, meta)
+}
+
+func resourceCloudflareWaitingRoomRead(d *schema.ResourceData, meta interface{}) error {
+	waitingRoomID := d.Id()
+	zoneID := d.Get("zone_id").(string)
+
+	return resourceCloudflareWaitingRoomReadFromWaitingRoomId(d, meta, zoneID, waitingRoomID)
+}
+
+func resourceCloudflareWaitingRoomUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	waitingRoomID := d.Id()
+	zoneID := d.Get("zone_id").(string)
+
+	waitingRoom := buildWaitingRoom(d)
+
+	_, err := client.ChangeWaitingRoom(context.Background(), zoneID, waitingRoomID, waitingRoom)
+
+	if err != nil {
+		name := d.Get("name").(string)
+		return fmt.Errorf("Error updating waiting room %q: %s", name, err)
+	}
+
+	return resourceCloudflareWaitingRoomRead(d, meta)
+}
+
+func resourceCloudflareWaitingRoomDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	waitingRoomID := d.Id()
+	zoneID := d.Get("zone_id").(string)
+
+	err := client.DeleteWaitingRoom(context.Background(), zoneID, waitingRoomID)
+
+	if err != nil {
+		name := d.Get("name").(string)
+		return fmt.Errorf("Error updating waiting room %q: %s", name, err)
+	}
+
+	return nil
+}
+
+func resourceCloudflareWaitingRoomImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	idAttr := strings.SplitN(d.Id(), "/", 2)
+	var zoneID string
+	var waitingRoomID string
+	if len(idAttr) == 2 {
+		zoneID = idAttr[0]
+		waitingRoomID = idAttr[1]
+	} else {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"zoneID/waitingRoomID\" for import", d.Id())
+	}
+
```

Comment:
checkout https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_argo_tunnel.go#L84 for an example of using the `Import` method to call the `Read` instead of a custom method.

Reply:
Done